src/sentry/sentry_metrics/indexer/postgres/postgres_v2.py

```diff
@@ -121,66 +154,107 @@ def bulk_record(
+        config = get_ingest_config(
+            self._get_metric_path_key(strings.keys()), IndexerStorage.POSTGRES
+        )
         writes_limiter = writes_limiter_factory.get_ratelimiter(config)
 
-        new_use_case_id = REVERSE_METRIC_PATH_MAPPING[use_case_id]
-        with writes_limiter.check_write_limits(
-            UseCaseKeyCollection({new_use_case_id: db_write_keys})
-        ) as writes_limiter_state:
+        """
+        Changes to writes_limiter will happen in a separate PR.
+        For now, we are going to operate on the assumption that no custom use case ID
+        will enter this part of the code path. Therethere strings can only be one of the
+        follow 2 types:
+        {
+            "sessions" : {
+                org_id_1: ... ,
+                org_id_n: ... ,
+            }
+        }
+        {
+            "transactions" : {
+                org_id_1: ... ,
+                org_id_n: ... ,
+            }
+        }
+        """
+        use_case_id = next(iter(strings.keys()))
+        use_case_path_key = self._get_metric_path_key(strings.keys())
+        with writes_limiter.check_write_limits(db_write_keys) as writes_limiter_state:
             # After the DB has successfully committed writes, we exit this
             # context manager and consume quotas. If the DB crashes we
             # shouldn't consume quota.
             use_case_collection = writes_limiter_state.accepted_keys
             # TODO: later we will use the whole use case collection instead
             # of pulling out the key collection
-            filtered_db_write_keys = use_case_collection.mapping[new_use_case_id]
+            filtered_db_write_keys = use_case_collection.mapping[use_case_id]
             del db_write_keys
 
-            rate_limited_key_results = KeyResults()
+            rate_limited_key_results = UseCaseKeyResults()
             for dropped_string in writes_limiter_state.dropped_strings:
-                rate_limited_key_results.add_key_result(
-                    dropped_string.key_result,
+                key_result = dropped_string.key_result
+                rate_limited_key_results.add_use_case_key_result(
+                    UseCaseKeyResult(
+                        use_case_id, key_result.org_id, key_result.string, key_result.id
+                    ),
                     fetch_type=dropped_string.fetch_type,
                     fetch_type_ext=dropped_string.fetch_type_ext,
                 )
 
             if filtered_db_write_keys.size == 0:
                 return db_read_key_results.merge(rate_limited_key_results)
 
-            new_records = []
-            for write_pair in filtered_db_write_keys.as_tuples():
-                organization_id, string = write_pair
-                new_records.append(
-                    self._table(use_case_id)(organization_id=int(organization_id), string=string)
-                )
+            if use_case_path_key is UseCaseKey.PERFORMANCE:
```

Comment:
I think this is not the best way to change the behavior of the indexer depending on the config. 
https://github.com/getsentry/sentry/blob/master/src/sentry/sentry_metrics/configuration.py#L34

Your code really need to change its behavior whether or not the indexer supports multiple use case ids. 
While this, today, correspond to the indexer being the performance one, it is just a coincidence.

In order to make this more robust I think I would suggest you add a new flag in the indexer configuration that drives this behavior (whether to return use case ids or not) and change your condition to be based on that.